cmplibrary/src/main/java/com/sourcepoint/cmplibrary/ConsentLib.java

```diff
@@ -238,14 +321,38 @@ public void onMessageChoiceSelect(int choiceId, int choiceType) {
      */
     public void run() throws ConsentLibException.NoInternetConnectionException {
         onMessageReadyCalled = false;
-        if(webView == null) { webView = buildWebView(); }
-        webView.loadMessage(sourcePoint.messageUrl(encodedTargetingParams, encodedAuthId ,encodedPMId));
-        mCountDownTimer = getTimer(defaultMessageTimeOut);
-        mCountDownTimer.start();
+        renderMessage();
         setSharedPreference(IAB_CONSENT_CMP_PRESENT, true);
         setSubjectToGDPR();
     }
 
+    private void renderMessage() throws ConsentLibException.NoInternetConnectionException {
+        if(webView == null) { webView = buildWebView(); }
+        sourcePoint.getMessageParams(new OnLoadComplete() {
+            @Override
+            public void onSuccess(Object result) {
+                try{
+                    String  msgUrl = ((JSONObject) result).getString("url");
+                    webView.loadUrlWithException(msgUrl);
+                    mCountDownTimer = getTimer(defaultMessageTimeOut);
+                    mCountDownTimer.start();
```

Comment:
I think this timer needs to be outside `getMessageParams`. We start a timer when the SDK starts running and we stop it either when the message shows up or when there's no message to show.

Reply:
Agreed, I din't do with the timer yet, just copied past this code by now.